File: solutions/python/series/1/series.py
def slices(series, length):
    if len(series) == 0:
        raise ValueError("series cannot be empty")
    if length == 0:
        raise ValueError("slice length cannot be zero")
    if length < 0:
        raise ValueError("slice length cannot be negative")
    if length > len(series):
        raise ValueError("slice length cannot be greater than series length")
    return [series[i:i+length] for i in range(0, len(series)) if len(series[i:i+length]) == length]

File: solutions/ruby/series/1/series.rb
class Series
  def initialize(series)
    @series = series.chars
  end

  def slices(slice)
    raise ArgumentError.new() if (slice <= 0 or slice > @series.length)
    (@series.length - slice + 1).times.map { |n| @series.drop(n).take(slice).join }
  end
end